### ISSUE TITLE
"Update" from 3.5.1 to 4.1.4

### ISSUE THREAD

**Issue by filmfacts** [bwssytems/ha-bridge#482]
I'm using 3.5.1 quite a while and can turn on, off or dim my homematic-actors. Also I can turn on or turn off my Yamaha RX-V775. Amazon Echo recognizes my devices and it works fine.

But if I copy the 4.1.4 (I tested the 4.0.1 with the same issue) into my ha-bridge directory, stop the 3.51 and start the 4.1.4 ... I can't control my devices. They show up in the GUI, I can edit the entries, but I can't turn them on or off. They simply don't react.

Do I have to enter my devices from scratch when switching vom 3.5.x to 4.x?

**Comment by rafale77** [comment]
Interesting. I have experienced the a similar problem with previous upgrades as well.
Though the devices configurations seem to be kept on the bridge UI after the upgrade, Amazon discovers them as new devices. I ended up with each device showing up twice on the echo device list with half of them inactive... Had to reconfigure all my groups.

**Comment by tuicemen** [comment]
Did you tell Alexa to discover devices after firing up 4.1.4?
As rafale77 stated you"ll have them show up twice in the Alexa app then delete the inactive ones.

**Comment by bwssytems** [comment]
I have an idea why that may occur. The ha-bridge generates a user String for the client side that is different after every start up. Just forget the devices and re-discover.

**Comment by rafale77** [comment]
You mean at every startup of the bridge? I am seeing this only after an upgrade. Have restarted the bridge many times without having this problem. When one has a dozen groups with over a hundred devices... it would be nice not to have to do this.

**Comment by tuicemen** [comment]
only after a update

**Comment by filmfacts** [comment]
It is nothing about the Echo. The devices won't work, if I click on the buttons in the web-gui of HA-Bridge even! I can rediscover the devices in Echo as many times as I want ... it will change nothing,

**Comment by bwssytems** [comment]
@filmfacts So, once you start it, it is unresponsive? Will need to see your log.

**Comment by dmsims** [comment]
When I did the same upgrade you have to edit each bridge device

Select the device Type from the dropdown menu and save

**Comment by bwssytems** [comment]
That is correct if you created them as Custom and did not select the Device Type or Map Type  previously.

**Comment by filmfacts** [comment]
Fantastic! That was the solution. After choosing HTTP_device everthing worked as expected. I just had to edit the entries for my Yamaha RX-V775 to switch it on and off oder change inputs, as the body-arguments weren't recognized by the new HA-Bridge-version.

**Comment by bwssytems** [comment]
What issues are there for the body arguments?

**Comment by filmfacts** [comment]
They were empty! In 3.5.1 everything works, after starting 4.1.4 they were empty.

**Comment by bwssytems** [comment]
Ahh, they are under the legacy fields, ha-bridge does not auto-populate the new fields, but if they are blank and the legacy fields are populated, it will use those.

**Comment by bwssytems** [comment]
Are there still issues?
